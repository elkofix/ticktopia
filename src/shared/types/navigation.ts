export interface NavigationItem {
    label: string
    href: string
    icon: React.ElementType
    roles: Role[]
    priority?: number 
}
