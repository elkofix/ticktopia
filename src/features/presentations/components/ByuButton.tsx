'use client';
import { useRouter } from 'next/navigation';

interface BuyButtonProps {
  href: string;
}

export function BuyButton({ href }: BuyButtonProps) {
  const router = useRouter();

  return (
    <button
      data-testid="buy-button"
      onClick={() => router.push(href)}
      className="bg-violet hover:bg-brand text-white py-4 px-12 text-lg font-bold rounded-full transition-colors duration-300 shadow-lg"
    >
      COMPRAR BOLETOS
    </button>
  );
}
