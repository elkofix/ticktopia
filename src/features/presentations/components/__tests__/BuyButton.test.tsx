import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { useRouter } from 'next/navigation';
import { BuyButton } from '../ByuButton';

// Mock the useRouter hook
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));

describe('BuyButton', () => {
  const mockPush = jest.fn();
  const testHref = '/checkout/event-123';

  beforeEach(() => {
    // Reset all mocks before each test
    jest.clearAllMocks();
    // Setup the mock router
    (useRouter as jest.Mock).mockReturnValue({
      push: mockPush,
    });
  });

  it('renders correctly with the correct text', () => {
    render(<BuyButton href={testHref} />);
    
    const button = screen.getByRole('button', { name: /COMPRAR BOLETOS/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('bg-violet');
    expect(button).toHaveClass('hover:bg-brand');
  });

  it('navigates to the correct href when clicked', () => {
    render(<BuyButton href={testHref} />);
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(mockPush).toHaveBeenCalledTimes(1);
    expect(mockPush).toHaveBeenCalledWith(testHref);
  });

  it('has the correct styling classes', () => {
    render(<BuyButton href={testHref} />);
    
    const button = screen.getByRole('button');
    expect(button).toHaveClass('bg-violet');
    expect(button).toHaveClass('hover:bg-brand');
    expect(button).toHaveClass('text-white');
    expect(button).toHaveClass('py-4');
    expect(button).toHaveClass('px-12');
    expect(button).toHaveClass('text-lg');
    expect(button).toHaveClass('font-bold');
    expect(button).toHaveClass('rounded-full');
    expect(button).toHaveClass('transition-colors');
    expect(button).toHaveClass('duration-300');
    expect(button).toHaveClass('shadow-lg');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<BuyButton href={testHref} />);
    expect(asFragment()).toMatchSnapshot();
  });
});