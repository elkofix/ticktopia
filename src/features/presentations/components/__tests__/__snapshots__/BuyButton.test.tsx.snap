// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="bg-violet hover:bg-brand text-white py-4 px-12 text-lg font-bold rounded-full transition-colors duration-300 shadow-lg"
  >
    COMPRAR BOLETOS
  </button>
</DocumentFragment>
`;
